Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/Asset/AssetEventChannel.tsx

```diff
@@ -0,0 +1,194 @@
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  04/19/2022 - Gabriel Santos
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Application, OpenXDA } from '@gpa-gemstone/application-typings';
+import { PhaseSlice, MeasurmentTypeSlice } from '../Store/Store'
+import Table from '@gpa-gemstone/react-table';
+import { useSelector } from 'react-redux';
+import { LoadingIcon, ServerErrorIcon } from '@gpa-gemstone/react-interactive';
+
+declare var homePath: string;
+
+interface IProps { Asset: OpenXDA.Types.Asset }
+
+interface ChannelDetail { //TODO: Move to Gemstone
+    ID: number,
+    MeterID: string,
+    AssetID: string,
+    MeasurementTypeID: string,
+    MeasurementCharacteristicID: string,
+    PhaseID: string,
+    Name: string,
+    Adder: number,
+    Multiplier: number,
+    SamplesPerHour: number,
+    PerUnitValue: number,
+    HarmonicGroup: number,
+    Description: string,
+    Enabled: boolean,
+    ConnectionPriority: number,
+    MeterKey: string,
+    MeterName: string,
+    AssetKey: string,
+    AssetName: string,
+    MeasurementType: string,
+    MeasurementCharacteristic: string,
+    Phase: string,
+    Mapping: string,
+    SeriesTypeID: string,
+    SeriesType: string
+}
+
+
+const AssetEventChannelWindow = (props: IProps) => {
+
+    const [assetChannels, setAssetChannels] = React.useState<ChannelDetail[]>([]);
+
+    const pStatus = useSelector(PhaseSlice.Status) as Application.Types.Status;
+    const mtStatus = useSelector(MeasurmentTypeSlice.Status) as Application.Types.Status;
+    const [status, setStatus] = React.useState<Application.Types.Status>('idle');
+
+
+    const [sortField, setSortField] = React.useState<keyof (ChannelDetail)>('ID');
+    const [ascending, setAscending] = React.useState<boolean>(true);
+
+
+
+    React.useEffect(() => {
+        let channelHandle = getChannels();
+
+        Promise.all([channelHandle]).then(() => setStatus('idle'), () => setStatus('error'))
+
+        return () => {
+            if (channelHandle != null && channelHandle.abort != null)
+                channelHandle.abort();
+        }
+    }, [props.Asset]);
+
+    function getChannels(): JQuery.jqXHR<ChannelDetail[]> {
+        setStatus('loading')
+        return $.ajax({
+            type: "GET",
+            url: `${homePath}api/OpenXDA/Asset/${props.Asset.ID}/ConnectedChannels`,
+            contentType: "application/json; charset=utf-A",
+            dataType: 'json',
+            cache: true,
+            async: true
+        }).done((d: Array<ChannelDetail>) => {
```

Comment:
```suggestion
        }).then((d: Array<ChannelDetail>) => {
```
`done` get's called regardless of an error. `then` let's you specify two separate callbacks one for successful calls and one for failed calls. If you only specify one it'll not do anything on failed calls. The error Case is actually handled in Line 82 so just a single callback is fine.

Reply:
Change implemented looks a little different than suggested. The suggested solution results in type errors on the promise return. It seems like react wants the promise to end in a done for error handling reasons (at least, that is what the internet is telling me). Implemented solution has the following structure:
$.ajax().done().fail();

With the mentioned cases in line 82 moved to this structure for clarity.